WooCommerce/Classes/ViewModels/Order List/OrderListCellViewModel.swift

```diff
@@ -0,0 +1,74 @@
+import Foundation
+import Yosemite
+
+struct OrderListCellViewModel {
+    private let order: Order
+    private let orderStatus: OrderStatus?
+
+    private let currencyFormatter = CurrencyFormatter(currencySettings: ServiceLocator.currencySettings)
```

Comment:
❓  For memory conservation purposes, it might be worth it to store this in `OrderListViewModel` instead and we will just inject the instance here. In that way, there's only one `currencyFormatter` lying around. What do you think?

Reply:
That's a good point and it would for sure improve memory allocation. I have an alternative suggestion, that may help in some other places as well.

#### Suggestion 1
* `CurrencyFormatter` is now a `class`, but it doesn't need to be. It doesn't have methods that mutate internal state and has only one member, which is a `let` member (CurrencySettings). If we convert it to `struct` we won't need to store it as a member variable in `OrderListCellViewModel`, but instead create and use the formatter "inline". I believe it will be basically free to create a currency formatter like this.

```
var total: String? {
      return CurrencyFormatter().formatAmount(order.total, with: order.currency)
}
```

* Maybe set a default value for `currencySettings` in the init method of `CurrencyFormatter`, because I see that it is always created this way: `CurrencyFormatter(currencySettings: ServiceLocator.currencySettings)`

#### Suggestion 2
I think it would be nice to have it stored somewhere, similar to how date formatters are created only once in `DateFormatter+Woo`. Of course the difference is that currency formatter depends on the site settings and needs to be updated the site is updated / changed. But this way there will be only one instance and everyone can use it without creating it over and over again :-)

Reply:
Let me know what you think, @shiki.
I think "Suggestion 2" may be a bit more beneficial in a long term run, but would require significant changes to multiple classes.
On the other hand other suggestions are also beneficial and require less code :-)

Reply:
Thanks for the detailed suggestion. Suggestion 1 sounds better for me. It looks like we should be passing `CurrencySettings` around instead of `CurrencyFormatter`. 

But I believe this is something that we should discuss more as a team since there was a previous discussion related to `CurrencyFormatter` (p91TBi-3bP-p2) and I might be missing something important. 

So, let's keep what we have in here as-is for now. And maybe create an issue for your suggestion? 🙂

Reply:
Thank you for sharing the link to a previous discussion. I will create an issue so that everyone can chip in there :-)